### PR TITLE
feat: example app and `waku_destroy`

### PR DESCRIPTION
**Pull request by richard-ramos**

### CODE REVIEW

**Review by SionoiS — APPROVED** [review]
Very nice thanks!

Would it be useful to store the node config in `WakuNodehandle`, only on `Running` nodes allow call to `waku_destroy` then return an `Initialized` node?

File: waku-bindings/src/node/management.rs [hunk 1]
```diff
@@ -43,6 +43,18 @@ pub fn waku_new(config: Option<WakuNodeConfig>) -> Result<WakuNodeContext> {
     }
 }
 
+pub fn waku_destroy(ctx: &WakuNodeContext) -> Result<()> {
```
Comment (Ivansete-status) [inline waku-bindings/src/node/management.rs:46]: Maybe it worth adding a little comment above

**Review by Ivansete-status — APPROVED** [review]
LGTM! Very educational PR, thanks for it!
qq: why `waku_destroy` is needed? We don't have such concept in `nwaku-libwaku`.

**Comment by richard-ramos** [comment]
> qq: why `waku_destroy` is needed? We don't have such concept in `nwaku-libwaku`.

The reason why it is necessary is because in `waku_new` we are creating a [context](https://github.com/waku-org/nwaku/blob/master/library/libwaku.nim#L83) running its own [thread](https://github.com/waku-org/nwaku/blob/e692edf6c19ef8584093840444bb12fd812a6566/library/waku_thread/waku_thread.nim#L89), yet we're not stopping that thread nor releasing the resources allocated by the `Context` by calling [`stopWakuNodeThread`](https://github.com/waku-org/nwaku/blob/e692edf6c19ef8584093840444bb12fd812a6566/library/waku_thread/waku_thread.nim#L98) .

In https://github.com/waku-org/nwaku/pull/2499 I added the `waku_destroy` function which calls that function, so we can do a proper cleanup of the resources used by waku when we no longer require a node. (Think of it as a graceful shutdown)

**Comment by Ivansete-status** [comment]
> > qq: why `waku_destroy` is needed? We don't have such concept in `nwaku-libwaku`.
> 
> The reason why it is necessary is because in `waku_new` we are creating a [context](https://github.com/waku-org/nwaku/blob/master/library/libwaku.nim#L83) running its own [thread](https://github.com/waku-org/nwaku/blob/e692edf6c19ef8584093840444bb12fd812a6566/library/waku_thread/waku_thread.nim#L89), yet we're not stopping that thread nor releasing the resources allocated by the `Context` by calling [`stopWakuNodeThread`](https://github.com/waku-org/nwaku/blob/e692edf6c19ef8584093840444bb12fd812a6566/library/waku_thread/waku_thread.nim#L98) .
> 
> In [waku-org/nwaku#2499](https://github.com/waku-org/nwaku/pull/2499) I added the `waku_destroy` function which calls that function, so we can do a proper cleanup of the resources used by waku when we no longer require a node. (Think of it as a graceful shutdown)

Wonderful! Thanks for that!

**Pull request merged**